velox/dwio/common/ScanSpec.h

```diff
@@ -403,6 +407,31 @@ class ScanSpec {
   std::vector<std::string> flatMapFeatureSelection_;
 };
 
+template <typename F>
+void ScanSpec::visit(const Type& type, F&& f) {
+  f(type, *this);
+  switch (type.kind()) {
+    case TypeKind::ROW:
```

Comment:
this can be simplified using a generic loop that works for all types

```
for (auto i = 0; i < type.size(); ++i) {
  ...visit type->childAt(i)
}
```

Reply:
Here we need to loop over the `ScanSpec` child nodes, and we only have a way from `ScanSpec` to `Type` but not reverse